haystack_experimental/components/extractors/llm_metadata_extractor.py

```diff
@@ -240,28 +253,80 @@ def from_dict(cls, data: Dict[str, Any]) -> "LLMMetadataExtractor":
+    def run(self, documents: List[Document], page_range: Optional[List[Union[str, int]]] = None):
         """
         Extract metadata from documents using a Language Model.
 
+        If `page_range` is provided, the metadata will be extracted from the specified range of pages. This component
+        will split the documents into pages and extract metadata from the specified range of pages. The metadata will be
+        extracted from the entire document if `page_range` is not provided.
+
+        The original documents will be returned  updated with the extracted metadata.
+
         :param documents: List of documents to extract metadata from.
+        :param page_range: A range of pages to extract metadata from. For example, page_range=['1', '3'] will extract
+                           metadata from the first and third pages of each document. It also accepts printable range
+                           strings, e.g.: ['1-3', '5', '8', '10-12'] will extract metadata from pages 1, 2, 3, 5, 8, 10,
+                           11, 12.
+                           If None, metadata will be extracted from the entire document for each document in the
+                           documents list.
         :returns:
             A dictionary with the keys:
-            - "documents": List of documents with extracted metadata.
+            - "documents": The original list of documents updated with the extracted metadata.
             - "errors": A dictionary with document IDs as keys and error messages as values.
         """
-        errors = {}
+
+        errors: Dict[str, Any] = {}
+
         for document in documents:
-            prompt_with_doc = self.builder.run(input_text=document.content)
-            result = self.llm_provider.run(prompt=prompt_with_doc["prompt"])
-            llm_answer = result["replies"][0]
-            if self.is_valid_json_and_has_expected_keys(expected=self.expected_keys, received=llm_answer):
-                extracted_metadata = json.loads(llm_answer)
-                for k in self.expected_keys:
-                    document.meta[k] = extracted_metadata[k]
+
+            if not document.content:
+                logger.warning(f"Document {document.id} has no content. Skipping metadata extraction.")
+                continue
+
+            if self.expanded_range or page_range: # extract metadata from a specific range of pages
+
+                # override the page range if provided
+                if page_range:
+                    self.expanded_range = expand_page_range(page_range)
+
+                if not self.expanded_range:
+                    msg = f"Page range {self.expanded_range} invalid"
+                    raise ValueError(msg)
+
+                splitter = DocumentSplitter(split_by="page", split_length=1)
+                pages = splitter.run(documents=[document])
+                content = [p.content + "\n" for idx, p in enumerate(pages["documents"]) if idx in self.expanded_range]
```

Comment:
We should reinsert the page break and not a newline and the idx starts at 0 whereas the `expanded_range` seems to start at 1. So this would send the wrong pages, shifted all by 1. So change to 
```suggestion
                content = [p.content + "\f" for idx, p in enumerate(pages["documents"]) if (idx + 1) in self.expanded_range]
```